pynest/nest/lib/hl_api_connection_helpers.py

```diff
@@ -54,24 +55,30 @@ def _process_conn_spec(conn_spec):
         raise TypeError("conn_spec must be a string or dict")
 
 
-def _process_syn_spec(syn_spec, conn_spec, prelength, postlength, connect_np_arrays):
+def _process_syn_spec(syn_spec, conn_spec, prelength, postlength, use_connect_arrays):
     """Processes the synapse specifications from None, string or dictionary to a dictionary."""
     if syn_spec is None:
+        # for use_connect_arrays, return "static_synapse" by default
+        if use_connect_arrays:
+            return {"synapse_model": "static_synapse"}
+
         return syn_spec
+
     rule = conn_spec['rule']
+
     if isinstance(syn_spec, str):
         return kernel.SLILiteral(syn_spec)
```

Comment:
Why is this case returning a literal, while all other cases return a dictionary, e.g. on line 61?

Reply:
Dictionary is not necessary on line 61 "static_synapse" would be fine... should I change it?
Also line 65 is None, so not a dict either.

Reply:
I think code is more readable if a function returns the same type in all cases, with `None` as the one exception to signal special cases. Since some other cases further down must return dicts, I would suggest to always return a dict.

Reply:
Roger, I'll update the code